rosbag2_transport/src/rosbag2_transport/topic_filter.cpp

```diff
@@ -218,6 +225,45 @@ bool TopicFilter::take_topic(
         return false;
       }
     }
+  } else {
+    // action topic
```

Comment:
Please check for `is_action_topic` explicitly
```suggestion
  } else if (is_action_topic) {
```